Sources/MapboxNavigation/NavigationMapView+Annotations.swift

```diff
@@ -0,0 +1,520 @@
+                      category: .navigationUI)
+        }
+    }
+    
+    func updateIntersectionAnnotations(with routeProgress: RouteProgress) {
+        guard !routeProgress.routeIsComplete else {
+            removeIntersectionAnnotations()
+            return
+        }
+        var featureCollection = FeatureCollection(features: [])
+        
+        let stepProgress = routeProgress.currentLegProgress.currentStepProgress
+        let intersectionIndex = stepProgress.intersectionIndex
+        let stepIntersections = stepProgress.intersectionsIncludingUpcomingManeuverIntersection
+        
+        for intersection in stepIntersections?.suffix(from: intersectionIndex) ?? [] {
+            if let feature = intersectionFeature(from: intersection) {
+                featureCollection.features.append(feature)
+            }
+        }
+        
+        let style = mapView.mapboxMap.style
+        
+        do {
+            let sourceIdentifier = NavigationMapView.SourceIdentifier.intersectionAnnotationsSource
+            if style.sourceExists(withId: sourceIdentifier) {
+                try style.updateGeoJSONSource(withId: sourceIdentifier, geoJSON: .featureCollection(featureCollection))
+            } else {
+                var source = GeoJSONSource()
+                source.data = .featureCollection(featureCollection)
+                try style.addSource(source, id: sourceIdentifier)
+            }
+            
+            let layerIdentifier = NavigationMapView.LayerIdentifier.intersectionAnnotationsLayer
+            guard !style.layerExists(withId: layerIdentifier) else { return }
+            
+            var shapeLayer = SymbolLayer(id: layerIdentifier)
+            shapeLayer.source = sourceIdentifier
+            shapeLayer.iconAllowOverlap = .constant(false)
+            shapeLayer.iconImage = .expression(Exp(.get) {
+                "imageName"
+            })
+            
+            let layerPosition = layerPosition(for: layerIdentifier)
+            try style.addPersistentLayer(shapeLayer, layerPosition: layerPosition)
+        } catch {
+            Log.error("Failed to perform operation while adding intersection signals with error: \(error.localizedDescription).",
+                      category: .navigationUI)
+        }
+    }
+    
+    private func intersectionFeature(from intersection: Intersection) -> Feature? {
+        var properties: JSONObject? = nil
+        if intersection.railroadCrossing == true {
+            properties = ["imageName": .string(ImageIdentifier.railroadCrossing)]
+        }
+        if intersection.yieldSign == true {
+            properties = ["imageName": .string(ImageIdentifier.yieldSign)]
+        }
+        if intersection.stopSign == true {
+            properties = ["imageName": .string(ImageIdentifier.stopSign)]
+        }
+        if intersection.trafficSignal == true {
+            properties = ["imageName": .string(ImageIdentifier.trafficSignal)]
+        }
```

Comment:
After check with Directions team, it seems yes. When a railroad crossing is controlled by a stop sign, it will set both the `stopSign` and `railroadCrossing` to `true`.  I updated the priority sequence of the intersection signals, when it has both  `stopSign` and `railroadCrossing`, it will show the  `railroadCrossing` sign.